tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/PublicClientApplicationTests.cs

```diff
@@ -326,6 +326,47 @@ public void AcquireTokenSameUserTest()
             }
         }
 
+        [TestMethod]
+        public void AcquireTokenWithDefaultRedirectURITest()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                harness.HttpManager.AddInstanceDiscoveryMockHandler();
+                PublicClientApplication app = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                                            .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
+                                                                            .BuildConcrete();
+                //Validate legacy default uri
+                Assert.AreEqual(app.AppConfig.RedirectUri, "urn:ietf:wg:oauth:2.0:oob");
+
+                app = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                                            .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
+                                                                            .WithHttpManager(harness.HttpManager)
+                                                                            .WithTelemetry(new TraceTelemetryConfig())
+                                                                            .WithDefaultRedirectUri()
+                                                                            .BuildConcrete();
+
+                //Validate new default redirect uri
+#if DESKTOP
+                Assert.AreEqual(app.AppConfig.RedirectUri, "https://login.microsoftonline.com/common/oauth2/nativeclient");
+#elif NET_CORE
+                Assert.AreEqual(app.AppConfig.RedirectUri, "https://localhost");
+#endif
+                MsalMockHelpers.ConfigureMockWebUI(

```

Comment:
I'd remove this second part of the unit test test and modify some of the integration tests instead.

Reply:
makes sense to remove the second part however I dont understand what you mean by "modify some of the integration tests instead". Are you saying that I need to add this check to some integration tests? Are unit tests not sufficient to test this scenario?